src/nvim/option.c

```diff
@@ -2030,7 +2030,13 @@ void set_init_1(void)
     } else {
       free(p_enc);
       p_enc = save_enc;
+      /* mb_init() failed. Call it again for the saved encoding */
+      mb_init();
```

Comment:
I think I follow now. If `set_init_1` fails to set the encoding, it sets `p_enc` back to the default encoding, but it doesn't call `mb_init()` to actually *apply the default (utf8) encoding* because it assumes it was already done at startup...?

Would you mind a quick debugging step to see what `enc_locale()` is returning in your case? Here are quick steps:

- start gbd in "TUI" mode: `gdb -tui build/bin/nvim`
- set a breakpoint at line 1987: `break option.c:1987`
- use the `n` command to step over that line
- use the `p` command to see the value of `p`: `p p`
```

What is the value of `p`?